### PR TITLE
Added custom error class and removed/refactored endpoints affected by new MR release

### PR DESCRIPTION
**Pull request by mattmanley**
### Description:
This PR adds a custom error class for the package. It also removes one method that uses an endpoint that got deleted (GET /project/:id/children) and changes another method that uses a deprecated endpoint (GET /projects/find )

### Potential Impact:
This PR will change the way errors are being caught for the API endpoints. 

### Unit Test Approach:
Tested the HTTP error handling by passing an ID that doesn't exist to "get_challenge_by_id" method. 

### Test Results:
The request failed, as expected: 
```
>>> import maproulette
>>> config = maproulette.Configuration()
>>> api = maproulette.Api(config)
>>> api.get_challenge_by_id('999999999999999')
/Users/matthewmanley/.local/share/virtualenvs/maproulette-python-client/lib/python3.6/site-packages/urllib3/connectionpool.py:1004: InsecureRequestWarning: Unverified HTTPS request is being made to host 'maproulette.org'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning,
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/Users/matthewmanley/Documents/github/maproulette-python-client/maproulette/api/api.py", line 109, in get_challenge_by_id
    response = self.server.get(endpoint=f"/challenge/{challenge_id}")
  File "/Users/matthewmanley/Documents/github/maproulette-python-client/maproulette/api/maproulette_server.py", line 30, in get
    raise HttpError(e) from None
maproulette.api.errors.HttpError: 404 Client Error: Not Found for url: https://maproulette.org/api/v2/challenge/999999999999999

```

### CODE REVIEW

File: maproulette/api/errors.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+"""This module contains the set of MapRoulette's exceptions"""
+
+
+class MapRouletteBaseException(Exception):
+    """MapRoulette Base Exception"""
```
Comment (danielduhh) [inline maproulette/api/errors.py:5]: Were you planning on making additional changes here? Each child exception inherits from the base, however --  MapRouletteBaseException doesn't actually do anything

File: maproulette/api/maproulette_server.py [hunk 2]
```diff
@@ -102,7 +157,20 @@ def delete(self, endpoint):
...
             response.raise_for_status()
         except requests.exceptions.HTTPError as e:
-            logging.critical(str(e))
+            if e.response.status_code == 401:
+                raise UnauthorizedError(
+                    message='The user is not authorized to make this request',
+                    status=e.response.status_code,
+                    payload=e.response
+                ) from None
+            elif e.response.status_code == 404:
```
Comment (danielduhh) [inline maproulette/api/maproulette_server.py:166]: What happens if neither 401 or 404 errors are thrown ?

File: maproulette/api/errors.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import json
+"""This module contains the set of MapRoulette's exceptions"""
+
+
+class MapRouletteBaseException(Exception):
+    """MapRoulette Base Exception"""
+    def __init__(self, message, status, payload=None):
+        self.message = message
+        self.status = status
```
Comment (danielduhh) [inline maproulette/api/errors.py:9]: This should be optional as well for non http errors. In the future, we can add extra linting for each api request to verify that the required props are given. For example, when creating a challenge via:
 
```python3
api.create_challenge_from_model
```
if "name" isn't supplied in the body, we can throw an MissingParameters that only contains a message.

File: maproulette/api/maproulette_server.py (hunk 2 above)
Comment (mattmanley) [reply maproulette/api/maproulette_server.py:166]: @danielduhh Just added a final step that will throw a generic HTTP error if not 401 or 404

**Pull request merged**